exporter/sentryexporter/factory.go

```diff
@@ -0,0 +1,56 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sentryexporter
+
+import (
+	"context"
+
+	"github.com/open-telemetry/opentelemetry-collector/component"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+)
+
+const (
+	typeStr = "sentry"
+)
+
+// Factory is the factory for the Sentry Exporter
+type Factory struct {
+}
+
+// Type gets the type of the Exporter config created by this factory
+func (f *Factory) Type() configmodels.Type {
+	return typeStr
+}
+
+// CreateDefaultConfig creates the default configuration for the Sentry Exporter
+func (f *Factory) CreateDefaultConfig() configmodels.Exporter {
+	return &Config{
+		ExporterSettings: configmodels.ExporterSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr,
+		},
+	}
+}
+
+// CreateTraceExporter creates a trace exporter based on the Sentry config
+func (f *Factory) CreateTraceExporter(ctx context.Context, params component.ExporterCreateParams,
+	config configmodels.Exporter) (component.TraceExporter, error) {
+
+	sentryConfig := config.(*Config)
```

Comment:
I see other exporters do exactly this. I need to read more of the exporters code base, in case you already did, I ask:

Is there any guarantee that `config` is always of type `*Config`?

If not, this is more robust (avoids a runtime panic):

```suggestion
	sentryConfig, ok := config.(*Config)
	if !ok {
		return nil, fmt.Errorf("unexpected config type: %T", config)
	}
```

(runnable example https://play.golang.org/p/Izk3YuHcqLD)